services/director-v2/src/simcore_service_director_v2/api/routes/health.py

```diff
@@ -1,13 +1,30 @@
 import datetime
+from typing import Annotated
 
-from fastapi import APIRouter
+from fastapi import APIRouter, Depends
 from models_library.api_schemas_directorv2.health import HealthCheckGet
+from servicelib.rabbitmq import RabbitMQClient
+
+from ...modules.rabbitmq import get_rabbitmq_client_from_request
+
+
+class HealthCheckError(RuntimeError):
+    """Failed a health check"""
+
 
 router = APIRouter()
 
 
 @router.get("/", response_model=HealthCheckGet)
-async def check_service_health():
+async def check_service_health(
+    rabbitmq_client: Annotated[
+        RabbitMQClient, Depends(get_rabbitmq_client_from_request)
+    ]
+):
+    if not rabbitmq_client.healthy:
+        msg = "RabbitMQ client is in a bad state!"
+        raise HealthCheckError(msg)
```

Comment:
Q: does this mean the container will restart? meaning director-v2 will be rebooted automatically if this happens?

Reply:
correct :+1:

Reply:
OK. SO I would like to make a suggestion here. Please add a message that is shared between all the services which use this pattern so that we can monitor restarts. This way we can investigate.
You would also want to setup some monitoring query for greyly or an alert at this point.

Reply:
done :+1: